src/sorted_run_iterator.rs

```diff
@@ -115,44 +196,57 @@ impl<'a> SortedRunIterator<'a> {
         })
     }
 
-    pub(crate) fn find_iter_from_key(
-        key: &[u8],
-        sorted_run: &'a SortedRun,
-    ) -> Iter<'a, SsTableHandle> {
-        match sorted_run.find_sst_with_range_covering_key_idx(key) {
-            None => sorted_run.ssts.iter(),
-            Some(idx) => sorted_run.ssts[idx..].iter(),
-        }
+    async fn advance_current_iter(&mut self) -> Result<(), SlateDBError> {
+        self.current_iter = match self.sorted_run_iter.next() {
+            None => None,
+            Some(h) => Some(
+                SstIterator::new(
+                    h,
+                    self.table_store.clone(),
+                    self.blocks_to_fetch,
+                    self.blocks_to_buffer,
+                    self.cache_blocks,
+                )
+                .await?,
+            ),
+        };
+        Ok(())
     }
 }
 
-impl<'a> KeyValueIterator for SortedRunIterator<'a> {
+impl<'a, H: AsRef<SsTableHandle>> KeyValueIterator for SortedRunIterator<'a, H> {
     async fn next_entry(&mut self) -> Result<Option<RowEntry>, SlateDBError> {
         loop {
             if let Some(iter) = &mut self.current_iter {
                 if let Some(kv) = iter.next_entry().await? {
                     return Ok(Some(kv));
+                } else {
+                    self.advance_current_iter().await?;
                 }
-                self.current_iter = match self.sorted_run_iter.next() {
-                    None => None,
-                    Some(h) => Some(
-                        SstIterator::new(
-                            h,
-                            self.table_store.clone(),
-                            self.blocks_to_fetch,
-                            self.blocks_to_buffer,
-                            self.cache_blocks,
-                        )
-                        .await?,
-                    ),
-                };
             } else {
                 return Ok(None);
             }
         }
     }
 }
 
+impl<'a, H: AsRef<SsTableHandle>> SeekToKey for SortedRunIterator<'a, H> {
+    async fn seek(&mut self, next_key: &Bytes) -> Result<(), SlateDBError> {
+        loop {
+            if let Some(iter) = &mut self.current_iter {
+                if iter.range_covers_key(next_key) {
+                    iter.seek(next_key).await?;
+                    return Ok(());
+                } else {
+                    self.advance_current_iter().await?;
+                }
+            } else {
+                return Ok(());
+            }
+        }
+    }
+}
+
 #[cfg(test)]
 mod tests {
```

Comment:
would be good to add some coverage for seek (e.g. seek within the current sst, seek across ssts, seek out of range)

Reply:
Yes, good call. I added a test case and fixed a bug in the implementation.